sway-fmt-v2/src/items/item_fn.rs

```diff
@@ -1,15 +1,165 @@
 use crate::{
-    fmt::{Format, FormattedCode, Formatter},
-    FormatterError,
+    fmt::{Format, FormattedCode, Formatter, FormatterError},
+    utils::bracket::{CurlyBrace, Parenthesis},
 };
-use sway_parse::ItemFn;
+use std::fmt::Write;
+use sway_parse::{token::Delimiter, CodeBlockContents, FnArg, FnArgs, FnSignature, ItemFn};
+use sway_types::Spanned;
 
 impl Format for ItemFn {
     fn format(
         &self,
-        _formatted_code: &mut FormattedCode,
+        formatted_code: &mut FormattedCode,
+        formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        self.fn_signature.format(formatted_code, formatter)?;
+        Self::open_curly_brace(formatted_code, formatter)?;
+        self.body
+            .clone()
+            .into_inner()
+            .format(formatted_code, formatter)?;
+        Self::close_curly_brace(formatted_code, formatter)?;
+
+        Ok(())
+    }
+}
+
+// For now this just pushes the char
+impl CurlyBrace for ItemFn {
+    fn open_curly_brace(
+        line: &mut FormattedCode,
+        _formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        line.push(Delimiter::Brace.as_open_char());
```

Comment:
I will need to double-check but shouldn't we also alter the `shape`?

Reply:
It's possible, I haven't added tests yet

Reply:
This has to do with the above in matching the `ItemBraceStyle`